src/test/scala/org/apache/spark/sql/delta/schema/SchemaEnforcementSuite.scala

```diff
@@ -253,6 +253,19 @@ trait AppendSaveModeTests extends BatchWriterTest {
       }
     }
   }
+
+  equivalenceTest("ensure schema mismatch error message contains table ID") {
+    disableAutoMigration {
+      withTempDir { dir =>
+        spark.range(10).write.append(dir)
+        val e = intercept[AnalysisException] {
+          spark.range(10).withColumn("part", 'id + 1).write.append(dir)
+        }
+        assert(e.getMessage.contains("schema mismatch detected"))
+        assert(e.getMessage.contains("Table ID:"))
```

Comment:
I think it would be good to check for the actual ID here, not just the Table ID string. You can get that by doing DeltaLog.forTable(spark, path).tableId

Reply:
Thanks. Added actual table ID in the test.